src/aiida_quantumespresso/workflows/pdos.py

```diff
@@ -484,10 +490,14 @@ def inspect_nscf(self):
             return self.exit_codes.ERROR_SUB_PROCESS_FAILED_NSCF
 
         if self.ctx.serial_clean:
+            # if scf was run in this workchain,
             # we no longer require the scf remote folder, so can clean it
-            cleaned_calcs = clean_workchain_calcs(self.ctx.workchain_scf)
-            if cleaned_calcs:
-                self.report(f"cleaned remote folders of SCF calculations: {' '.join(map(str, cleaned_calcs))}")
+            # otherwise, we assume the user wants to keep scf remote folder
+            # as it was not generated in this workchain and might be used for other calculations
+            if self.should_run_scf():
```

Comment:
I think we can just join both conditions on line 492, i.e.:

```python
        if self.ctx.serial_clean and self.should_run_scf():
```

Perhaps the `self.should_run_scf()` is even unnecessary, and checking if `scf` is in the inputs is more direct/clearer?

```python
        if self.ctx.serial_clean and 'scf' in self.inputs:
```